yt_dlp/extractor/manyvids.py

```diff
@@ -1,114 +1,99 @@
-            webpage = self._download_webpage(url, video_id)
+        info = traverse_obj(
+            self._download_json(f'https://www.manyvids.com/bff/store/video/{video_id}', video_id),
+            ('data', {dict})) or {}
 
-        info = self._search_regex(
-            r'''(<div\b[^>]*\bid\s*=\s*(['"])pageMetaDetails\2[^>]*>)''',
-            webpage, 'meta details', default='')
-        info = extract_attributes(info)
-
-        player = self._search_regex(
-            r'''(<div\b[^>]*\bid\s*=\s*(['"])rmpPlayerStream\2[^>]*>)''',
-            webpage, 'player details', default='')
-        player = extract_attributes(player)
+        video_urls = traverse_obj(
+            self._download_json(f'https://www.manyvids.com/bff/store/video/{video_id}/private', video_id),
+            ('data', {dict})) or {}
 
         video_urls_and_ids = (
-            (info.get('data-meta-video'), 'video'),
-            (player.get('data-video-transcoded'), 'transcoded'),
-            (player.get('data-video-filepath'), 'filepath'),
-            (self._og_search_video_url(webpage, secure=False, default=None), 'og_video'),
+            (traverse_obj(video_urls, ('teaser', 'filepath')), 'preview'),
+            (video_urls.get('transcodedFilepath'), 'transcoded'),
+            (video_urls.get('filepath'), 'filepath'),
         )
 
-        def txt_or_none(s, default=None):
-            return (s.strip() or default) if isinstance(s, str) else default
-
-        uploader = txt_or_none(info.get('data-meta-author'))
-
-        def mung_title(s):
-            if uploader:
-                s = re.sub(rf'^\s*{re.escape(uploader)}\s+[|-]', '', s)
-            return txt_or_none(s)
-
-        title = (
-            mung_title(info.get('data-meta-title'))
-            or self._html_search_regex(
-                (r'<span[^>]+class=["\']item-title[^>]+>([^<]+)',
-                 r'<h2[^>]+class=["\']h2 m-0["\'][^>]*>([^<]+)'),
-                webpage, 'title', default=None)
-            or self._html_search_meta(
-                'twitter:title', webpage, 'title', fatal=True))
-
-        title = re.sub(r'\s*[|-]\s+ManyVids\s*$', '', title) or title
-
-        if any(p in webpage for p in ('preview_videos', '_preview.mp4')):
+        title = info.get('title')
+        uploader = traverse_obj(info, ('model', 'displayName'))
+        description = info.get('description')
+        likes = parse_count(info.get('likes'))
+        views = parse_count(info.get('views'))
+        thumbnail = url_or_none(info.get('thumbnail')) or url_or_none(info.get('screenshot'))
+        release_timestamp = parse_iso8601(info.get('launchDate'))
+        duration = parse_duration(info.get('videoDuration'))
+        tags = [t.get('label') for t in info.get('tagList')]
+
+        # If the video formats JSON only contains a teaser object, then it is a preview
+        if video_urls.get('teaser') and not video_urls.get('filepath'):
             title += ' (Preview)'
```

Comment:
I'm honestly happy either way, it's not a big change

Reply:
@octopusknives I'm going to put a compromise here and keep `(Preview)` in the title in addition to the warning. I think it's fair that the previews should have some different type of title to indicate they are previews rather than the full length.

Reply:
more importantly, the `id` needs to have a `-preview` suffix added so that yt-dlp doesn't erroneously add entries to the user's download archive